app/controllers/carts_controller.rb

```diff
@@ -1,22 +1,11 @@
 class CartsController < ApplicationController
   def index
-    if session[:cart_id] == nil
-      @cart = nil
-    else
-      @cart = ShoppingCart.find(session[:cart_id])
-    end
+    find_or_create_cart
```

Comment:
Good job. You can probably add this as an `before_action` if called from two methods (index and show)